lib/linalg/Matrix.h

```diff
@@ -491,6 +491,107 @@ class Matrix
         int this_row,
         Vector& other) const;
 
+    /**
+     * @brief Multiplies two matrices element-wise.
+     *
+     * @pre numRows() == other.numRows()
+     * @pre numColumns() == other.numColumns()
```

Comment:
Should there also be prerequisites about whether the matrices are distributed? Also for some other new functions in this class.